### DISCUSSION TITLE
Can i import from Elsa to Bpmn and vice versa ?

### DISCUSSION THREAD

**Discussion by tacek** [elsa-workflows/elsa-core#267] [Q&A]
Hi!
I am write about Elsa.
There are 3 questions that interest me.
1) Can i import from Elsa to Bpmn and vice versa?
2) Loading a process from JSON to C#, that is to say - there is a description in json and i need to import it at start app, is there any way to do this through code?
3) Is there a standard solution to start a process by POST request or do i need to write a separate web application to handle POST request?

I would really appreciate your help!
Thank you.

**Comment by sfmskywalker** [comment]
Hi @tacek !

1. Not at the moment. I do intend to investigate the possibilities in this area. Or if anyone else is interested in looking into this they would be more than welcome.
1. Yes, you can write & read JSON files representing workflow definitions. See [Sample09](https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/Sample09/Program.cs) for an example that reads JSON from an embedded resource and then uses that JSON to deserialize & run a workflow.
1. Yes, all you need to do is create your ASP.NET Core application and register some Elsa services, most importantly the HTTP activity services. Please have a look at [this guide](https://elsa-workflows.github.io/elsa-core/docs/guides-hello-world-http). Although that guide shows you how to implement the workflow in C# code, it works the same way when defining the workflow using the dashboard application. For more information on setting up the dashboard, please refer to [this guide](https://elsa-workflows.github.io/elsa-core/docs/guides-dashboard). It demonstrates exactly your question.

**Comment by sfmskywalker** [comment]
Closing this issue, but feel free to reopen if you have follow-up questions!

**Comment by sfmskywalker** [comment]
Short answer: as of today, Elsa still does not ship BPMN import/export as a supported feature.

For the current codebase, `elsa-core` explicitly states that it is not a BPMN 2.0 engine and does not parse BPMN XML (`doc/security-assessment/elsa-core-profile.md:37-40`). The roadmap still lists BPMN interoperability as a future item rather than a shipped capability (`ROADMAP.md:89`, `ROADMAP.md:181`).

For the other two parts of your question, workflow definitions can be imported and exported through the current workflow definition endpoints:
- import: `workflow-definitions/import` and `workflow-definitions/{definitionId}/import` (`src/modules/Elsa.Workflows.Api/Endpoints/WorkflowDefinitions/Import/Endpoint.cs:38-42`)
- export: `/workflow-definitions/{definitionId}/export` and `/bulk-actions/export/workflow-definitions` (`src/modules/Elsa.Workflows.Api/Endpoints/WorkflowDefinitions/Export/Endpoint.cs:15-19`)

So the current supported interchange story is Elsa JSON/ZIP workflow definitions, not BPMN. And for HTTP-triggered workflows, Elsa supports that through its HTTP activities and server hosting model rather than requiring a separate custom app just to receive POST requests.

Since the original question was already answered and the current status is clear, I'm closing this discussion as resolved.
